File: Reservations/Reservations.Domain/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Reservations.Domain
{
    public class Event : AuditableEntity
    {
        public Event()
        {
            Questions = new HashSet<Question>();
            EventOccurences = new HashSet<EventOccurrence>();
        }
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        //Stored as ticks
        public long Length { get; set; }
        public int EventTypeId { get; set; }
        public EventType EventType { get; set; }
        public ICollection<Question> Questions { get; set; }
        public ICollection<EventOccurrence> EventOccurences { get; set; }
    }
}

File: reservation-frontend/src/app/models/event.ts
import { EventOccurrence } from "./eventOccurrence"
import { EventType } from "./eventType"
import { Question } from "./question"

export class Event {
    public id!: number
    public name!: string
    public directorName!: string
    public description!: string
    public length!: number
    public eventTypeId!: number
    public eventType!: EventType
    public questions!: Question[]
    public eventOccurrences!: EventOccurrence[]
}